app/forms/recovery_code_form.rb

```diff
@@ -25,9 +25,9 @@ def valid_recovery_code?
     recovery_code_generator.verify(code)
   end
 
-  def result
+  def extra_analytics_attributes
     {
-      success: success,
+      method: 'recovery code',
```

Comment:
Yes, it is important. What I don't see is how making it a constant reduces the chances of someone changing it. It's only referred to once in the whole app here.